### ISSUE TITLE
[Lang Enhancement Proposal] Implement Compiler Support for C-code

### ISSUE THREAD

**Issue by Luna-Klatzer** [Para-Lang/Para#72]
**Affected area by change:**
- Language Syntax: no
- Language Library (Parac Base Library): no
- Compiler: yes
- Pre-Processor: yes
- Documentation: yes

**Short Introduction and Summary**

Implement the Compiler-support for the C language and headers, so that in the compiled code the C headers and source files are usable.

**Is your request related to a problem? Please describe.**

After #32, the idea of Para-C being fully based on the C syntax was scratched. This was due to syntax limitations, which will bring more harm than good, as new syntax could simplify the implementation of the features. 

This though now creates the problem that Para-C would be an isolated language, and other environments based on C will not be able to be included anymore, and as such new code would have to be written to use Para-C for specialized usages and replace the otherwise usable C libraries.

**Describe the solution you'd like**

Implement the Compiler-support for the C language and headers, so that in the compiled code the C headers and source files are usable. This would also mean that types and functions are usable, even if not in the original C syntax. 

Summary of Implementation Ideas:
- That means that all `typedef` types and structs will be native C types and not allow for the advanced Para-C handling. Though this should not require additional checking in the code, as all types will be per default incompatible, though for native ones the standard convert functions (C -> PBL) may be used. 
- `PblAny_T` will not be allowed for C types, as it strictly requires Para-C types.
- Functions may be called, though exceptions are disabled and the convert functions (PBL -> C) need to be used.
- Pre-Processor macros will be also included if possible and handled, though limited (This may be its own issue if the implementation is at that point)

**Describe alternatives you've considered**

None.

**Additional context**

- [x] #32

**Comment by Luna-Klatzer** [comment]
Dropping this idea, as the language makes more sense to be standalone and the effort of implementing C is not worth it.

May add C extensions in the future like those available in Python, but for now it doesn't make any sense.
